### PR TITLE
ARROW-13544 [Java]: Remove APIs that have been deprecated for long (Changes to ArrowBuf)

### PR DESCRIPTION
**Pull request by liyafan82**
See https://issues.apache.org/jira/browse/ARROW-13544

According to the discussion in https://github.com/apache/arrow/pull/10864#issuecomment-895707729, we want to split the task into multiple parts. 

This PR is for the changes related to `ArrowBuf`

### CODE REVIEW

File: java/vector/src/test/java/org/apache/arrow/vector/util/DecimalUtilityTest.java [hunk 1]
```diff
@@ -41,7 +41,8 @@ public void testSetLongInDecimalArrowBuf() {
       ) {
         int [] intValues = new int [] {Integer.MAX_VALUE, Integer.MIN_VALUE, 0};
         for (int val : intValues) {
-          buf.clear();
```
Comment (emkornfield) [inline java/vector/src/test/java/org/apache/arrow/vector/util/DecimalUtilityTest.java:44]: given these changes, I think maybe we should keep clear() and remove all the rest.  adding more calls for readerIndex and writeIndex seems like it is moving in the wrong direction.

Comment (liyafan82) [reply java/vector/src/test/java/org/apache/arrow/vector/util/DecimalUtilityTest.java]: sounds good. I have revised the PR to preserve the method.

**Comment by emkornfield** [comment]
CC @jacques-n not sure if there is a current point of contact at dremio, but I'm incline to start removing the deprecated methods here that were used as a crutch for some prior cleanup.  I'll merge early next week unless there are objections.

**Comment by jacques-n** [comment]
> CC @jacques-n not sure if there is a current point of contact at dremio, but I'm incline to start removing the deprecated methods here that were used as a crutch for some prior cleanup. I'll merge early next week unless there are objections.

I'd suspect that @vvellanki, @laurentgo, @StevenMPhillips and @pravindra should be able to provide better insights on these objections.

**Comment by laurentgo** [comment]
@emkornfield you should proceed. We had enough time to modify our own code, and if we still have some references left, that's on us.

**Pull request closed without merging**